2. Switch NLog configuration to config file

rurounijones opened the pull request:
Switch NLog configuration to config file

Switch the configuration of NLog to a config file from the current
in-code configuration.

This makes it much easier to tailor logging to the requirements of the
developer or user.

I have chosen to use a separate "NLog.config" file by default from the
various options available to us (See
https://github.com/nlog/nlog/wiki/Configuration-file#configuration-file-locations
for the full list) as I think it is the simplest and easiest to
understand by most people.

This commit includes a config file and hard-coded defaults that almost,
but not quite, replicates the old hard-coded config. Differences are:

1. We delegate log rotation to NLog itself.

This means that in its current configuration it will log up to 200MB
of data (100MB in the current log and 100MB in the old log)

This is unlike the existing setup which will log to an unlimited
filesize until the client is restarted and then only creates an "old"
file if the existing one is over 100MB on application startup. I don't
expect this to have any real-world impact and the file sizes, number of
old log files and other settings can now be changed as the user wishes.

2. There is no hard-coded "Debug" LogLevel set on app debug mode.

For VS users the copy settings will only copy the config file if it is
newer than the one in the build directory which means developers can
modify their logging settings as they please (e.g. set the log level to
debug mode once and it will persist until the project logging file is
updated which should not happen often).

Of course, since this is all in a config file, we can change all of this
as we please.

Note that the existing hard-coded config for the SRS client was
incorrectly calling the fileTarget directly instead of the async
wrapper. This was fixed as part of this commit, the SRS server was
correctly calling the async wrapper.

### Testing

Tested by running Client and server with and without config files and making sure they still created the log files.

rurounijones commented on DCS-SR-Client/DCS-SR-Client.csproj:
```diff
@@ -17,6 +17,7 @@
     <IsWebBootstrapper>false</IsWebBootstrapper>
     <NuGetPackageImportStamp>
     </NuGetPackageImportStamp>
+    <TargetFrameworkProfile />
```
I have no idea why this got switched. VS 2019 establishing dominance of ordering I guess.

ciribob commented:
What happens if the file is missing?

 Does this need to be added to the installer?

Thanks for sorting!

rurounijones commented:
> What happens if the file is missing?

No logging takes place

> Does this need to be added to the installer?

I don't know how the installer works or what changes need to be made there. Does something need to be done when adding new files?

ciribob commented:
Ah right, could you change it so this file overrides sensible defaults that are built in rather than it not working without this file?

The same code can be applied to the server then as well

rurounijones commented:
Changed the hardcoded value to match the config file.
* This changed the rollover as described in the commit message in the hardcoded as well.
* SRS Client was not using the async wrapper originally, but is now.

Added warnings to the config file and code to remind people to keep them in sync.

Tested by running Client and server with and without config files and making sure they still created the log files.
